src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1481,6 +1490,42 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
 
         return null;
     }
+    
+    /**
+     * Add log event for request failure with a cause. 
+     */
+    private void addLogEventForRequestFailureWithCause(final String eventName, final AuthenticationRequest request, 
+            final AuthenticationException authenticationException) {
+        ClientAnalytics.logEvent(
+                eventName,
+                new InstrumentationPropertiesBuilder(request, authenticationException).build());
+    }
+    
+    /**
+     * @return True if access token is returned from result, false otherwise. 
+     */
+    private boolean isAccessTokenReturned(final AuthenticationResult authResult) {
+        return authResult != null && !StringExtensions.IsNullOrBlank(authResult.getAccessToken());
+    }
+    
+    /**
+     * If access token redeem with family refresh token fails, should fall back to MRRT if existed. 
+     */
+    private AuthenticationResult fallbackToMRRT(final AuthenticationRequest authRequest, final RefreshItem refreshItem) 
```

Comment:
That actually makes code less readable, and you've got this feedback already from 2 different people that this method doesn't make sense. if you want to make it better create method that will do
getToken ()
makeNetworkCall()
And call it twice